spec/integration/rpc_contract.rb

```diff
@@ -26,6 +26,10 @@ def call
   end
 
   class Router < Protein::Router
+    config(around_processing: ->(block, name, klass) do
```

Comment:
Why have this configuration on the router level, not in the server itself? The server already supports configuring it. And since we're pretty much using a DSL here, I think it's more customary in Ruby to define an `around` callback as:

```ruby
around do
  # do before stuff
  yield
  # do after stuff
end
```

or

```ruby
around do |thing|
  # do before stuff
  thing.call
  # do after stuff
end
```

I'm not sure lambdas don't have any drawbacks when used instead of procs.

Reply:
That was my initial idea as well, but as it turns out `Server` is not being used in every context: `Router` is common for `HTTP` and `AMQP`, while `Server` works only with `AMQP` adapter (the other one exposes `rack` middleware, but doesn't support `serve` method that `Server` expects from adapter). 

When it comes to configuration method I totally agree. I did it this way to keep consistency with `on_worker_boot` configuration option on `Server` level (actually I more or less copied it here :P).

I'd treat your comment as a sanity check: Protein became chaotic over time, and deviated from reasonable design. If we're to use it, we should consider fixing those design flaws. If we're to fix the implementation, maybe it's good moment to reevaluate if we want to keep it in our tech stack...